### PR TITLE
Disallow duplicate json keys

### PR DESCRIPTION
**Pull request by janl**

### CODE REVIEW

**Comment by rnewson** [comment]
please remove whitespace changes

**Comment by rnewson** [comment]
+1 once ws changes are removed

**Comment by janl** [comment]
@davisp @rnewson the WS changes are a separate commit

**Comment by davisp** [comment]
@janl The issue is that the new whitespace is also wrong as its using two space indents instead of our standard 4. If you update to have correct indentation I'm fine leaving it as per SOP.

File: src/chttpd/src/chttpd_auth_cache.erl [hunk 1]
```diff
@@ -45,7 +45,7 @@ get_user_creds(_Req, UserName) when is_binary(UserName) ->
             Props;
         UserProps when is_list(UserProps) ->
             couch_auth_cache:add_roles(Props,
-	        couch_util:get_value(<<"roles">>, UserProps))
+            couch_util:get_value(<<"roles">>, UserProps))
```
Comment (davisp) [inline src/chttpd/src/chttpd_auth_cache.erl]: This is missing an extra two indents since its a continuation line.

**Comment by janl** [comment]
sorry this slipped, fixed in latest push

**Pull request merged**